presto-main/src/main/java/com/facebook/presto/sql/analyzer/MaterializedViewQueryOptimizer.java

```diff
@@ -0,0 +1,325 @@
+    {
+        Expression rewriteSortKey = (Expression) process(node.getSortKey(), context);
+        return new SortItem(
+                rewriteSortKey,
+                node.getOrdering(),
+                node.getNullOrdering());
+    }
+
+    @Override
+    protected Node visitSimpleGroupBy(SimpleGroupBy node, MaterializedViewQueryOptimizerContext context)
+    {
+        ImmutableList.Builder<Expression> rewriteSimpleGroupBy = ImmutableList.builder();
+        for (Expression column : node.getExpressions()) {
+            Expression rewriteColumn = (Expression) process(column, context);
+            rewriteSimpleGroupBy.add(rewriteColumn);
+        }
+        return new SimpleGroupBy(rewriteSimpleGroupBy.build());
+    }
+
+    public static final class MaterializedViewQueryOptimizerContext
+    {
+        private Table materializedViewTable;
+        private Query originalSqlQuery;
+        private Map<String, String> baseToViewColumnMap;
+        private Optional<Relation> baseQueryRelation;
+
+        public MaterializedViewQueryOptimizerContext(
+                Table materializedViewTable,
+                Query originalSqlQuery)
+        {
+            this.materializedViewTable = materializedViewTable;
+            this.originalSqlQuery = originalSqlQuery;
+            baseToViewColumnMap = new HashMap<>();
+            baseQueryRelation = Optional.empty();
+            createBaseToViewColumnMap();
+        }
+
+        public void createBaseToViewColumnMap()
+        {
+            QuerySpecification originalSqlQueryBody = (QuerySpecification) originalSqlQuery.getQueryBody();
+
+            if (originalSqlQueryBody.getFrom().isPresent()) {
+                this.baseQueryRelation = originalSqlQueryBody.getFrom();
+            }
+
+            Select baseTableSelectFields = originalSqlQueryBody.getSelect();
+
+            for (SelectItem baseTableSelectColumn : baseTableSelectFields.getSelectItems()) {
+                if (baseTableSelectColumn instanceof SingleColumn) {
+                    String baseColumnName = ((SingleColumn) baseTableSelectColumn).getExpression().toString();
+                    Optional<Identifier> viewOptionalDerivedName = ((SingleColumn) baseTableSelectColumn).getAlias();
+                    String viewDerivedColumnName = baseColumnName;
+
+                    if (viewOptionalDerivedName.isPresent()) {
+                        viewDerivedColumnName = viewOptionalDerivedName.get().getValue();
+                    }
+
+                    baseToViewColumnMap.put(baseColumnName, viewDerivedColumnName);
+                }
+            }
+        }
+
+        public Table getMaterializedViewTable()
+        {
+            return materializedViewTable;
+        }
+
+        public String getViewColumnName(String baseColumnName)
+        {
+            checkState(baseToViewColumnMap.containsKey(baseColumnName), "Missing column name in the conversion map: " + baseColumnName);
```

Comment:
I don't think we should call checkState. This would throw an exception and would lead to query failure.
If in any case we can not optimize the query, we should continue to use the base query. 
Lets also add test cases to make sure base query remains as it is for unsupported cases.